Test fx

landersanmi opened the pull request.

landersanmi committed:
añadido testFX

landersanmi committed:
test fxButton added

Prueba para comprobar si aumenta el test coverage

landersanmi committed:
miniError

landersanmi committed:
monocle added

landersanmi committed:
properties added

erikberter committed:
Prueba Random

erikberter committed:
Update UserDetailControllerTest.java

erikberter committed:
Prueba de añadir a github actions

erikberter committed:
Update UserDetailControllerTest.java

erikberter committed:
Tests funcionando de TestFx

landersanmi committed:
Añadidos testFx en UserDetailController

erikberter committed:
Merge branch 'main' into testFX

landersanmi committed:
actualizado con tokens

landersanmi committed:
test DeleteUser funciona

landersanmi committed:
Testeada clase userDetailController

landersanmi committed:
Añadido test de CurrentBalanceController

erikberter commented on DeustockClient/pom.xml:
```diff
@@ -71,7 +71,8 @@
 					<reuseForks>true</reuseForks>
-					<argLine>${surefireArgLine} -Xmx1024m -noverify --enable-preview</argLine>
+					<!--suppress UnresolvedMavenProperty -->
+                    <argLine>${surefireArgLine} -Xmx1024m -noverify --enable-preview</argLine>
```
¿Que has cambiado aquí?

erikberter commented on DeustockClient/src/main/java/es/deusto/deustock/client/controllers/UserDetailController.java:
```diff
@@ -57,16 +71,13 @@ public void setParams(HashMap<String, Object> params) {
+    public void deleteUser(){
+        if(gateway.deleteUser(mainController.getToken())){
+            mainController.setUser(null);
+            mainController.loadAndChangeScene(
```
Esta linea es pequeña, se puede poner en una sola linea.

landersanmi replied:
Puesta en 1 linea

erikberter commented on DeustockClient/src/main/java/es/deusto/deustock/client/controllers/UserDetailController.java:
```diff
@@ -90,25 +101,31 @@ private void initRoot(){
     	if(!succesfullyReseted) {
-    	    // TODO handle with exception
-    		System.out.println("No se ha podido resetear");
+            throw new Exception("No se ha podido resetear");
```
No uses exception genericas.

landersanmi replied:
Cambiado a RuntimeException

landersanmi committed:
Empty commit for workflows

erikberter commented on DeustockClient/src/test/java/es/deusto/deustock/client/controllers/UserDetailControllerTest.java:
```diff
@@ -0,0 +1,213 @@
+        HashMap<String, Object> params = new HashMap<>();
+        params.put("username", "Test");
+
+        when(mockGateway.getUser(anyString())).thenReturn(user);
+        when(mockGateway.deleteUser(any())).thenReturn(true);
+        doNothing().when(mockMainController).setUser(any());
+        doNothing().when(mockMainController).loadAndChangeScene(anyString());
+
+        controller.setMainController(mockMainController);
+        controller.setDeustockGateway(mockGateway);
+        Platform.runLater( () -> controller.setParams(params) );
+
+        // When
+        controller.deleteUser();
+
+    }
+
+    @Test
+    void testDeleteUserOnNonExistentUser() {
+        // Given
+        User user = new User();
+        user.setUsername("username");
+        HashMap<String, Object> params = new HashMap<>();
+        params.put("username", "Test");
+
+
+
+        when(mockGateway.getUser(anyString())).thenReturn(user);
+        when(mockGateway.deleteUser(any())).thenReturn(false);
+
+        doNothing().when(mockMainController).setUser(any());
+        doNothing().when(mockMainController).loadAndChangeScene(anyString());
+
+        controller.setMainController(mockMainController);
+        controller.setDeustockGateway(mockGateway);
+        Platform.runLater( () -> controller.setParams(params) );
+
+        // When
+        controller.deleteUser();
+    }
```
Cuando no haya fallos pon assertDoesNotThrow

landersanmi replied:
cambiado a assertDoesNotThrow

erikberter commented on DeustockClient/src/test/java/es/deusto/deustock/client/controllers/CurrentBalanceControllerTest.java:
```diff
@@ -0,0 +1,122 @@
+
+        // When
+        Platform.runLater( () -> controller.setParams(params));
+    }
+
+    @Test
+    void testOpenOnNullUser() {
+        // Given
+        HashMap<String, Object> params = new HashMap<>();
+        params.put("grfhrt", "Tetrhtst");
+
+        when(mockGateway.getUser(anyString())).thenReturn(null);
+        controller.setDeustockGateway(mockGateway);
+
+        // When
+        Platform.runLater( () -> controller.setParams(params) );
+
+    }
+
+
+    @Test
+    void testRefreshStocks() {
+        // Given
+        User user = new User();
+        user.setUsername("Test2");
+        HashMap<String, Object> params = new HashMap<>();
+        params.put("username", "Test");
+
+        StockHistory sh1 = new StockHistory().setActualPrice(20.3).setOpenPrice(3.5).setSymbol("BB").setAmount(4).setOperation(OperationType.SHORT);
+        List<StockHistory> sh = new ArrayList<StockHistory>();
+        sh.add(sh1);
+
+        when(mockGateway.getUser(anyString())).thenReturn(user);
+        when(mockGateway.getBalance(anyString())).thenReturn(9999.0);
+        when(mockGateway.getHoldings(anyString())).thenReturn(sh);
+        controller.setDeustockGateway(mockGateway);
+
+        // When
+        Platform.runLater( () -> controller.setParams(params));
+    }
```
Cuando el codigo deba ejecutar bien, pon simplemente assertDoesNotThrow. No dejes test sin assert

erikberter commented on .github/workflows/maven.yml:
```diff
@@ -62,7 +62,7 @@ jobs:
 
     - name: Build with Maven (Client)
-      run: mvn --file DeustockClient/pom.xml --batch-mode --update-snapshots verify
+      run: xvfb-run mvn --file DeustockClient/pom.xml --batch-mode --update-snapshots verify
```
¿Has añadido esto en todos los workflow?

erikberter committed:
Reordenamiento de archivos de vistas y modificaciones en Holding view

landersanmi committed:
Update codeql-analysis.yml

landersanmi committed:
Update sonar.yml

landersanmi committed:
Update pom.xml

landersanmi committed:
Update UserDetailController.java

landersanmi committed:
Update CurrentBalanceControllerTest.java

landersanmi committed:
Update UserDetailControllerTest.java

landersanmi committed:
Small Errors

landersanmi committed:
Exception modified

erikberter committed:
Test Controller Login

landersanmi committed:
ChangeUserDetailControllertTest añadido

landersanmi committed:
Merge branch 'testFX' of https://github.com/futotta-risu/DeuStock into testFX

erikberter committed:
Merge branch 'main' into testFX

erikberter committed:
Corregido  POM

erikberter committed:
Merge branch 'main' into testFX

erikberter committed:
Prueba optimización de POM

erikberter committed:
Corregido LoginController

erikberter committed:
Corregido y optimizado test

erikberter committed:
Revertido intento de optimizacion de POM

landersanmi committed:
Diseño avanzado en UserDetailView y ChangeUserDetailView

erikberter committed:
RegisterController test

erikberter committed:
Prueba de correccion de error en test

aritzugazaga committed:
Hecho el primer test de la clases Stocks

landersanmi committed:
Diseño avanzado de vistas user terminado

landersanmi committed:
Cabezeras iguales en todas las vistas

aritzugazaga committed:
Test Search stockNotFound creado

aritzugazaga committed:
Clase stocklist test acabada

aritzugazaga committed:
Seguimiento en la creacion de los test de Stock

erikberter committed:
Merge branch 'main' into testFX

erikberter committed:
Corregido error Gateway repetido

erikberter committed:
Añadida etiqueta NotThreadSafe para evitar problemas de simultaneidad

erikberter committed:
Eliminados tamaños fijos

aritzugazaga committed:
Avance en tests

landersanmi committed:
Animaciones Añadidas a los botones

Login daba fallo al intentar cargar el css
Estando en home el tamaño de los botones  de ControlButtonController no cambia al hacer hover

aritzugazaga committed:
Operation controller acabado

aritzugazaga committed:
Clase StockDetailControllerTest acabada

landersanmi committed:
Merge branch 'main' into testFX

landersanmi committed:
Cambios en ChangeUserDetailControllerTest

Commit:
Home and Help controllers tested

Commit:
:,)

landersanmi committed:
Merge branch 'testFX' of https://github.com/futotta-risu/DeuStock into testFX

aritzugazaga committed:
avance stock

aritzugazaga committed:
Merge branch 'testFX' of https://github.com/futotta-risu/DeuStock into testFX

aritzugazaga committed:
Update OperationControllerTest.java

erikberter committed:
Correcciones algunas

erikberter committed:
Merge branch 'main' into testFX

erikberter committed:
Eliminados tags

erikberter committed:
Merge branch 'testFX' of https://github.com/futotta-risu/DeuStock into testFX

landersanmi committed:
Creados tests de integracion para DeustockGateway [STOCKS]

landersanmi committed:
Merge branch 'main' of https://github.com/futotta-risu/DeuStock into main

aritzugazaga committed:
Test stocks controllers hechos

erikberter committed:
ForkCount change

erikberter committed:
Merge branch 'testFX' of https://github.com/futotta-risu/DeuStock into testFX

erikberter committed:
Cambio en los forks

erikberter committed:
Intento de resolver errores

erikberter committed:
Update LoginControllerTest.java

erikberter committed:
Update LoginControllerTest.java

erikberter committed:
Update LoginControllerTest.java

erikberter committed:
Configurados tests

landersanmi committed:
Prueba IT

landersanmi committed:
Prueba 2 IT

landersanmi committed:
Prueba 2.2 IT

landersanmi committed:
Prueba 3 IT

landersanmi committed:
Prueba 3.2 IT

landersanmi committed:
Prueba 4 IT

erikberter committed:
Sistema funcional

erikberter committed:
Merge branch 'D01-Client_Style' into testFX

landersanmi committed:
Update maven.yml

landersanmi committed:
Update maven.yml

landersanmi committed:
Update maven.yml

erikberter committed:
Commit funcional

erikberter committed:
Merge branch 'main' into testFX

erikberter committed:
Corregido error menor en merge

erikberter committed:
Merge branch 'testFX' of https://github.com/futotta-risu/DeuStock into testFX

erikberter committed:
Cambios

landersanmi committed:
Documentacion añadida en cliente

erikberter committed:
Corregidos tests tras merge

erikberter committed:
Merge branch 'main' into testFX

erikberter committed:
Merge branch 'main' into testFX
